src/After.tsx

```diff
@@ -19,9 +19,9 @@ class Afterparty extends React.Component<any, any> {
     if (navigated) {
       window.scrollTo(0, 0);
       // save the location so we can render the old screen
+      // and keep the data so the render won't break
       this.setState({
         previousLocation: this.props.location,
-        data: undefined, // unless you want to keep it
```

Comment:
This can leave stale data around.

Reply:
I removed that line in order to render the last location components without any issues but then I added the _shouldComponentUpdate_ method, so it's not rendering again the whole tree hence I can add that line back.